mesonbuild/dependencies/base.py

```diff
@@ -1411,7 +1411,9 @@ def _detect_dep(self, name: str, modules: T.List[T.Tuple[str, bool]], components
                                       'Valid targets are:\n{}'.format(name, list(self.traceparser.targets.keys())))
 
         # Set dependencies with CMake targets
-        reg_is_lib = re.compile(r'^(-l[a-zA-Z0-9_]+|-pthread)$')
+        # recognise arguments we should pass directly to the linker
+        reg_is_lib = re.compile(r'(^-l[a-zA-Z0-9_\.]+|-pthread|-delayload:[a-zA-Z0-9_\.]+|[a-zA-Z0-9_]+\.(lib|dll))$')
```

Comment:
I think dll could be removed from here as well, but it probably doesn't cause any harm.

Reply:
Gotcha, done this as well. Thanks!